### DISCUSSION TITLE
Seeding does not behave as expected

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2629] [Q&A]
As I understand it, a seeder should simply insert some data into a specific table.

However, the following example code does not work:

```ts
import {EntityManager} from '@mikro-orm/core'
import {Seeder} from '@mikro-orm/seeder'
import {Entity} from './entity/Entity.entity'

export class DevSeeder extends Seeder {
  async run(em: EntityManager): Promise<void> {
    const a = em.create(Entity, {
      whatever: 5
    })
  }

  // em.persistAndFlush(a)
}
```

It does not insert any data in the respective table when running `mikro-orm seeder:run -c DevSeeder` _unless I uncomment `em.persistAndFlush(a)`_.

However, the [docs](https://mikro-orm.io/docs/next/seeding/) explicitly say that _[r]unning a seeder from the command line or programmatically will automatically call `flush` and `clear` after the `run` method has completed_.

It may be though that I misunderstood the docs and `clear()` actually deletes the data from memory (cache) and is (from one point of view) the opposite of `persist()`.

I suggest to update the docs (the note cited above) to add a note that if one wants to persist the data, they should explicitly persist it into the DB.

**Comment by B4nan** [comment]
you need to persist the entity, otherwise it can't be flushed. `em.create()` just creates entity instance, the EM is not aware of it. or use a factory as the other examples are showing, factory will persist those entities automatically

cc @Langstra this will need improvements either in the code or in the docs. that example cant do anything as `em.create()` won't persist the entity.

**Reply by Langstra** [reply]
Funny thing this. Calling the seeder from the command line should call the following methods in the following order. 

- `DatabaseSeedCommand.handler()`
- `SeedManager.seedString()`
- `SeedManager.seed()`

The last method calls `await this.orm.em.flush();      this.orm.em.clear();`

Which means it should flush I guess?

We could make it even more explicit by calling `em.persist()` after running a seeder or editing the documentation, but it seems unnecessary.

**Reply by B4nan** [reply]
This is not about how CLI or SeedManager does things, the code inside the seeder (which is an example from the docs) can't work on its own, as it does not persist the entity, just creates the instance - the EM is not aware of it, hence flush does nothing.

edit: I can imagine adding `Seeder.create()` that would do `em.create()` & `em.persist()`, or we could maybe implement support for autopersisting to `em.create()` that would be disabled by default (but enabled in the context of CLI or in general for seeder. or we could use a `Proxy` to tap into the `em.create()` in the seeder context, as there we definitely want to persist.
